pkg/edit/editor_test.go

```diff
@@ -37,6 +37,16 @@ func TestEditor_DoesCommandWithLeadingSpaceToHistory(t *testing.T) {
 	testCommands(t, f.Store /* no commands */)
 }
 
+func TestEditor_DoesCommandFilterBeforeAddToHistory(t *testing.T) {
+	f := setup(rc("edit:add-cmd-filters = [[_]{ put $false }]"))
```

Comment:
The test passes for the following incorrect implementations:

* Suppress history as long as `$edit:add-cmd-filters` is not empty
* Suppress history as long as function in `$edit:add-cmd-filters` has any output at all

Use a more complex filter function that can return true or false for different inputs.